components/equihash/tromp/equi_miner.c

```diff
@@ -649,50 +693,33 @@ nc++,       candidate(eq, tree_from_bid(bucketid, s0, s1));
 
 typedef struct {
   u32 id;
-  pthread_t thread;
   equi *eq;
 } thread_ctx;
 
-void barrier(pthread_barrier_t *barry) {
-  const int rc = pthread_barrier_wait(barry);
-  if (rc != 0 && rc != PTHREAD_BARRIER_SERIAL_THREAD) {
-//    printf("Could not wait on barrier\n");
-    pthread_exit(NULL);
-  }
-}
-
 void *worker(void *vp) {
   thread_ctx *tp = (thread_ctx *)vp;
   equi *eq = tp->eq;
 
-  if (tp->id == 0)
+//  if (tp->id == 0)
 //    printf("Digit 0\n");
-  barrier(&eq->barry);
   equi_digit0(eq, tp->id);
```

Comment:
Just keeping this code in sync
```suggestion
  if (tp->id == 0) {
    equi_clearslots(eq);
  }
  equi_digit0(eq, tp->id);
```